gobblin-api/src/main/java/gobblin/source/extractor/WatermarkSerializerHelper.java

```diff
@@ -0,0 +1,37 @@
+package gobblin.source.extractor;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+
+/**
+ * Provides default implementation for converting a {@link Watermark} to a {@link JsonElement}, and vice versa. The
+ * class uses <a href="https://code.google.com/p/google-gson/">GSON</a> to achieve this. This class provides a default
+ * way to serialize and de-serialize {@link Watermark}s, and is useful for implementing the {@link Watermark#toJson()}
+ * method.
+ */
+public class WatermarkSerializerHelper {
```

Comment:
It's not used anywhere. It just provides a default serialization technique for and users implementing a new Extractor or Source. It may become useful when using the complex watermarks in the Kafka Adapter.